lib/teleterm/daemon/daemon.go

```diff
@@ -126,19 +126,21 @@ func (s *Service) ResolveCluster(uri string) (*clusters.Cluster, error) {
 	return cluster, nil
 }
 
-// ResolveFullCluster returns full cluster information. It makes a request to the auth server.
+// ResolveFullCluster returns full cluster information. It makes a request to the auth server and includes
+// details about the cluster and logged in user
 func (s *Service) ResolveFullCluster(ctx context.Context, uri string) (*clusters.Cluster, error) {
 	cluster, err := s.ResolveCluster(uri)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	features, err := cluster.GetClusterFeatures(ctx)
+	clusterDetails, err := cluster.GetClusterDetails(ctx)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	cluster.Features = features
+	cluster.LoggedInUser = clusterDetails.LoggedInUser
+	cluster.Features = clusterDetails.Features
```

Comment:
I wouldn't mind but my reasoning is, because the name `cluster.GetClusterDetails`, it seems like it should return details. This can be reused elsewhere. If we chose to set them in `cluster.GetClusterDetails` then it would have to return a `Cluster` instead and should probably change the name to like... `AddClusterDetails` or something and that seems a bit weirder to me. 

I won't die on any hill, I opt for the way it is, but I'm open to any debate on the matter.

Reply:
> If we chose to set them in `cluster.GetClusterDetails` then it would have to return a `Cluster` instead and should probably change the name to like... `AddClusterDetails` or something and that seems a bit weirder to me.

Yeah, I'm not sure what the convention for this is; the only thing which comes to my mind is something like uhhhh `Enrich…`?

My argument for this would consist of two points. First, since we set those fields on `Cluster` anyway, we could let the struct itself set them rather than doing it from outside.

Second, I generally try to avoid doing things that _might_ be useful later unless I have reasons to believe that there will be a place to use them. So far I haven't heard about reusing those cluster details elsewhere and if we ever ended up needing them elsewhere, we could always refactor the existing code.

Ultimately this particular usage is also not something super important for me. What I care about more than that is what patterns we establish and avoiding adding stuff for which we don't have use at the moment. We can have a nuanced discussion under this PR but the next person who's going to need a similar object which requires "enriching" is going to copy what's already in the project.